design/ceph-mon-pv.md

```diff
@@ -0,0 +1,139 @@
+# Ceph monitor PV storage
+
+Currently all of the storage for Ceph monitors (data, logs, etc..) is provided
+using HostPath volume mounts. Supporting PV storage for Ceph monitors in
+environments with dynamically provisioned network block devices (AWS, GCE,
+etc...) will allow monitors to migrate without requiring the monitor state to be
+rebuilt. This is particularly useful in cases where nodes fails or are taken
+offline for maintenance or upgrades.
+
+There are two major components to this design. First, the cluster CRD schema is
+modified to accept a PVC template for defining the storage to be used by
+monitors. And second, monitor pod management will be handled by StatefulSets
+rather than Deployments.
+
+# Monitor storage
+
+The monitor specification in the CRD is updated to include a persistent volume
+claim template that is used to obtain PVs for the monitor data store.
+
+```go
+type MonSpec struct {
+	Count                int  `json:"count"`
+	PreferredCount       int  `json:"preferredCount"`
+	AllowMultiplePerNode bool `json:"allowMultiplePerNode"`
+	VolumeClaimTemplate  *v1.PersistentVolumeClaim
+}
+```
+
+Unlike the storage requirements of OSDs (xf: StorageClassDeviceSets) it is
+reasonable to use default requirements for monitors (e.g. 5-10 GB). This means
+that a minimal claim may be specified using only the requested storage class in
+which case default resource requests will be used.
+
+```yaml
+volumeClaimTemplates:
+- spec:
+    storageClassName: gp2-ebs
+...
+volumeClaimTemplates:
+- spec:
+    storageClassName: local-storage
+```
+
+It is also reasonable for monitors to use storage in the form of a directory on
```

Comment:
You are saying we will use our current implementation with dataDirHostPath when the template is omitted, right?

Reply:
I'm saying that if the template is omitted, then a local directory on host--equivalent to the current behavior--would be used. Whether that is hostpath based or a local pv that maps a local path is a separate issue, i think. Does that answer your question?